pkg/cableengine/syncer/syncer_test.go

```diff
@@ -335,9 +342,22 @@ func (t *testDriver) run() {
 		t.handledError <- err
 	})
 
+	Expect(submarinerv1.AddToScheme(kubeScheme.Scheme)).To(Succeed())
+
+	scheme := runtime.NewScheme()
+	Expect(submarinerv1.AddToScheme(scheme)).To(Succeed())
+
+	dynamicClient := fakeClient.NewSimpleDynamicClient(scheme)
+	restMapper := test.GetRESTMapperFor(&submarinerv1.Endpoint{})
+
 	client := fakeClientset.NewSimpleClientset()
 	t.gateways.GatewayInterface = client.SubmarinerV1().Gateways(namespace)
-	t.syncer = syncer.NewGatewaySyncer(t.engine, t.gateways, t.expectedGateway.Status.Version)
+	t.healthChecker, _ = healthchecker.New(&watcher.Config{
+		RestMapper: restMapper,
+		Client:     dynamicClient,
+		Scheme:     scheme,
+	}, namespace, "west")
```

Comment:
The syncer should take a `healthchecker.Interface` so you can create a mock/fake.

Reply:
Done.